context/context.go

```diff
@@ -0,0 +1,76 @@
+package context
+
+import (
+	gocontext "context"
+	"fmt"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Context interface {
+	gocontext.Context
+	WithGoContext(c gocontext.Context) Context
```

Comment:
Why would we replace the underlying context ? If anything, it would be a WithNewContext()